frameworks/elastic/src/main/dist/elasticsearch.yml

```diff
@@ -19,6 +19,18 @@ bootstrap.memory_lock: false
 metrics.statsd.host: {{STATSD_UDP_HOST}}
 metrics.statsd.port: {{STATSD_UDP_PORT}}
 
+{{#SHARD_ALLOCATION_AWARENESS_ENABLED}}
+node.attr.region: ${REGION}
+node.attr.zone: ${ZONE}
+cluster.routing.allocation.awareness.attributes: region,zone
+{{/SHARD_ALLOCATION_AWARENESS_ENABLED}}
+
+{{^SHARD_ALLOCATION_AWARENESS_ENABLED}}
+{{#CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTES}}
```

Comment:
If you don't have `SHARD_ALLOCATION_AWARENESS_ENABLED` and you do have `CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTES` set to `foo`, where do you set `node.attr.foo`? I would suggest deleting this `CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTES` block.

Reply:
I agree.

Just a note that deleting this would make it difficult for users to configure their own allocation awareness if for some reason `SHARD_ALLOCATION_AWARENESS_ENABLED` isn't exactly what they want (they'd have to use custom yaml blocks or the runtime cluster APIs). I'm guessing this scenario might be rare so not supporting it makes sense.